addons/web_editor/static/src/js/widgets/ace.js

```diff
@@ -562,12 +564,18 @@ var ViewEditor = Widget.extend({
         }).bind(this));
         if (errorFound) return $.Deferred().reject(errorFound);
 
-        var defs = [];
+        var mutex = new concurrency.Mutex();
         _.each(toSave, (function (_toSave, type) {
-            defs = defs.concat(_.map(_toSave, (type === 'xml' ? this._saveView : this._saveSCSS).bind(this)));
+            // Child views first as COW on parent will delete it
+            _toSave = _.sortBy(_toSave, 'id').reverse();
+            _.each(_toSave, function (session) {
+                mutex.exec(function () {
+                    return (type === 'xml' ? self._saveView(session) : self._saveSCSS(session));
+                });
+            });
         }).bind(this));
 
-        return $.when.apply($, defs).fail((function (session, error) {
+        return mutex.getUnlockedDef().fail((function (session, error) {
```

Comment:
Mmh in fact `getUnlockedDef` never fails. To get the same behavior as before, you have to... do stuff difficult to explain by writing  :cry:

Reply:
Maybe you can try to draw smthing @qsm-odoo  ? 

"one picture is worth a thousand words" ?

Reply:
Done, as discussed needed to `hack` the mutex as .fail() can't happen on mutex